src/meds_torch/data/components/multiwindow_pytorch_dataset.py

```diff
@@ -14,6 +14,131 @@
 from meds_torch.data.components.pytorch_dataset import PytorchDataset
 
 
+def fuse_window_data(windows_data: dict, windows_to_fuse: list[str], fused_window_name: str) -> dict:
+    """Fuse multiple windows into a single window, tracking the lengths of original windows.
+
+    Args:
+        windows_data (dict): Dictionary containing data from multiple windows to be fused
+        windows_to_fuse (list[str]): List of window names to fuse in specified order
+        fused_window_name (str): Name for the resulting fused window
+
+    Returns:
+        dict: Fused window data with length tracking information
+
+    Raises:
+        ValueError: If fusion configuration is invalid or data type is unsupported
+
+    Example:
+    >>> import torch
+    >>> # Create mock data with different types
+    >>> windows_data = {
+    ...     "pre": {
+    ...         # 1D tensor
+    ...         "static_values": torch.tensor([1.0, 2.0]),
+    ...         # 2D tensor (batch_size=2, seq_len=3, features=2)
+    ...         "embeddings": torch.ones(2, 3, 2),
+    ...         # List
+    ...         "codes": ["A", "B", "C"],
+    ...         # Scalar
+    ...         "scalar": 42
+    ...     },
+    ...     "post": {
+    ...         "static_values": torch.tensor([3.0, 4.0]),
+    ...         "embeddings": torch.ones(2, 2, 2) * 2,
+    ...         "codes": ["D", "E"],
+    ...         "scalar": 43
+    ...     }
+    ... }
+    >>> # Fuse windows
+    >>> fused = fuse_window_data(
+    ...     windows_data,
+    ...     windows_to_fuse=["pre", "post"],
+    ...     fused_window_name="fused"
+    ... )
+    >>> # Check lengths tracking
+    >>> fused["LENGTHS//static_values"]  # Two windows with lengths 2 and 2
```

Comment:
The comments here are a bit confusing as they aren't specific to static vs. dynamic?

Reply:
Clarified this in the docstring